lib/codecs/src/decoding/framing/chunked_gelf.rs

```diff
@@ -0,0 +1,650 @@
+        assert_eq!(frame, Some(Bytes::from(expected_message)));
+    }
+
+    #[rstest]
+    #[tokio::test]
+    async fn decode_unordered_messages(
+        two_chunks_message: ([BytesMut; 2], String),
+        three_chunks_message: ([BytesMut; 3], String),
+    ) {
+        let (mut two_chunks, two_chunks_expected) = two_chunks_message;
+        let (mut three_chunks, three_chunks_expected) = three_chunks_message;
+        let mut decoder = ChunkedGelfDecoder::default();
+
+        let frame = decoder.decode_eof(&mut three_chunks[2]).unwrap();
+        assert!(frame.is_none());
+
+        let frame = decoder.decode_eof(&mut two_chunks[0]).unwrap();
+        assert!(frame.is_none());
+
+        let frame = decoder.decode_eof(&mut three_chunks[0]).unwrap();
+        assert!(frame.is_none());
+
+        let frame = decoder.decode_eof(&mut two_chunks[1]).unwrap();
+        assert_eq!(frame, Some(Bytes::from(two_chunks_expected)));
+
+        let frame = decoder.decode_eof(&mut three_chunks[1]).unwrap();
+        assert_eq!(frame, Some(Bytes::from(three_chunks_expected)));
+    }
+
+    #[rstest]
+    #[tokio::test]
+    async fn decode_mixed_chunked_and_unchunked_messages(
+        unchunked_message: (BytesMut, String),
+        two_chunks_message: ([BytesMut; 2], String),
+    ) {
+        let (mut unchunked_message, expected_unchunked_message) = unchunked_message;
+        let (mut chunks, expected_chunked_message) = two_chunks_message;
+        let mut decoder = ChunkedGelfDecoder::default();
+
+        let frame = decoder.decode_eof(&mut chunks[1]).unwrap();
+        assert!(frame.is_none());
+
+        let frame = decoder.decode_eof(&mut unchunked_message).unwrap();
+        assert_eq!(frame, Some(Bytes::from(expected_unchunked_message)));
+
+        let frame = decoder.decode_eof(&mut chunks[0]).unwrap();
+        assert_eq!(frame, Some(Bytes::from(expected_chunked_message)));
+    }
+
+    #[rstest]
+    #[tokio::test(start_paused = true)]
+    #[traced_test]
+    async fn decode_timeout(two_chunks_message: ([BytesMut; 2], String)) {
+        let (mut chunks, _) = two_chunks_message;
+        let mut decoder = ChunkedGelfDecoder::default();
+
+        let frame = decoder.decode_eof(&mut chunks[0]).unwrap();
+        assert!(frame.is_none());
+        assert!(!decoder.state.lock().unwrap().is_empty());
+
+        // The message state should be cleared after a certain time
+        tokio::time::sleep(Duration::from_millis(DEFAULT_TIMEOUT_MILLIS + 1)).await;
+        assert!(decoder.state.lock().unwrap().is_empty());
+        assert!(logs_contain(
+            "Message was not fully received within the timeout window. Discarding it."
+        ));
+
+        let frame = decoder.decode_eof(&mut chunks[1]).unwrap();
+        assert!(frame.is_none());
+
+        tokio::time::sleep(Duration::from_millis(DEFAULT_TIMEOUT_MILLIS + 1)).await;
+        assert!(decoder.state.lock().unwrap().is_empty());
+        assert!(logs_contain(
```

Comment:
Is it okay to assert `log_contains` in these tests? Could not find any easy way of testing that behaviour without doing this